tests/framework/nodeplacement.go

```diff
@@ -53,11 +53,36 @@ func (f *Framework) TestNodePlacementValues() sdkapi.NodePlacement {
 }
 
 // PodSpecHasTestNodePlacementValues compares if the pod spec has the set of node placement values defined for testing purposes
-func (f *Framework) PodSpecHasTestNodePlacementValues(podSpec v1.PodSpec) bool {
+func (f *Framework) PodSpecHasTestNodePlacementValues(name string, podSpec v1.PodSpec) bool {
```

Comment:
I would split this into 3 functions
1. node selector check
2. anti affinity check
3. toleration check
1 you can probably do in operator_test.go something like this:
```go
for _, deploymentName := range []string{"cdi-apiserver", "cdi-deployment", "cdi-uploadproxy"} {
  deployment, err := f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Get(context.TODO(), deploymentName, metav1.GetOptions{})
  Expect(err).ToNot(HaveOccurred())
  Expect(deployment.Spec.Template.Spec.NodeSelector).To(Equal(framework.NodeSelectorTestValue))
 CheckAntiAffinity(...)
 CheckTolerations(...)
 ...
```